### PR TITLE
[Merged by Bors] - feat(algebra/order): unions of Ioc intervals

### PR DESCRIPTION
**Pull request by loefflerd**
Forward-port lemmas in `Data.Set.Intervals.Group` about disjointness of intervals, from mathlib3 PR [#18427](https://github.com/leanprover-community/mathlib/pull/18427)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Intervals/Group.lean [hunk 1]
```diff
@@ -5,12 +5,14 @@ Authors: Johannes Hölzl, Mario Carneiro, Patrick Massot, Yury Kudryashov, Rémy
 Ported by: Winston Yin
 
 ! This file was ported from Lean 3 source module data.set.intervals.group
-! leanprover-community/mathlib commit 18a5306c091183ac90884daa9373fa3b178e8607
+! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Intervals/Group.lean:8]: According to https://leanprover-community.github.io/mathlib-port-status/file/data/set/intervals/group?range=18a5306c091183ac90884daa9373fa3b178e8607..207cfac9fcd06138865b5d04f7091e46d9320432 this is the wrong SHA

Comment (loefflerd) [reply Mathlib/Data/Set/Intervals/Group.lean]: Copy-paste error, sorry! (That sha is the commit listed immediately below the last-synced commit in the port-status view, while it is the one above I meant to copy across.)

File: Mathlib/Data/Set/Intervals/Group.lean [hunk 2]
```diff
@@ -5,12 +5,14 @@ Authors: Johannes Hölzl, Mario Carneiro, Patrick Massot, Yury Kudryashov, Rémy
...
 ! This file was ported from Lean 3 source module data.set.intervals.group
-! leanprover-community/mathlib commit 18a5306c091183ac90884daa9373fa3b178e8607
+! leanprover-community/mathlib commit 740acc0e6f9adf4423f92a485d0456fc271482da
 ! Please do not edit these lines, except to modify the commit id
 ! if you have ported upstream changes.
 -/
 import Mathlib.Data.Set.Intervals.Basic
 import Mathlib.Algebra.Order.Group.Abs
+import Mathlib.Data.Set.Pairwise
+import Mathlib.Algebra.GroupPower.Lemmas
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Intervals/Group.lean:15]: This import order isn't the same as the mathlib3 order:
```suggestion
import Mathlib.Data.Set.Intervals.Basic
import Mathlib.Data.Set.Pairwise
import Mathlib.Algebra.Order.Group.Abs
import Mathlib.Algebra.GroupPower.Lemmas
```

(from https://github.com/leanprover-community/mathlib3port/blob/6e37b6d7464082351091ed0a9fb032205c21aae2/Mathbin/Data/Set/Intervals/Group.lean#LL11-L14C41)

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Pull request closed without merging**